qiita_db/commands.py

```diff
@@ -80,25 +80,25 @@ def get_optional(name):
         if optvalue is not None:
             infodict[value] = optvalue
 
-    emp_person_name_email = get_optional('emp_person_name')
-    if emp_person_name_email is not None:
-        emp_name, emp_email, emp_affiliation = emp_person_name_email.split(',')
-        infodict['emp_person_id'] = StudyPerson.create(emp_name.strip(),
-                                                       emp_email.strip(),
-                                                       emp_affiliation.strip())
-    lab_name_email = get_optional('lab_person')
-    if lab_name_email is not None:
-        lab_name, lab_email, lab_affiliation = lab_name_email.split(',')
-        infodict['lab_person_id'] = StudyPerson.create(lab_name.strip(),
-                                                       lab_email.strip(),
-                                                       lab_affiliation.strip())
+    with TRN:
+        emp_person_name_email = get_optional('emp_person_name')
+        if emp_person_name_email is not None:
+            emp_name, emp_email, emp_affiliation = \
+                emp_person_name_email.split(',')
+            infodict['emp_person_id'] = StudyPerson.create(
+                emp_name.strip(), emp_email.strip(), emp_affiliation.strip())
+        lab_name_email = get_optional('lab_person')
+        if lab_name_email is not None:
+            lab_name, lab_email, lab_affiliation = lab_name_email.split(',')
+            infodict['lab_person_id'] = StudyPerson.create(
+                lab_name.strip(), lab_email.strip(), lab_affiliation.strip())
 
-    pi_name_email = infodict.pop('principal_investigator')
-    pi_name, pi_email, pi_affiliation = pi_name_email.split(',', 2)
-    infodict['principal_investigator_id'] = StudyPerson.create(
-        pi_name.strip(), pi_email.strip(), pi_affiliation.strip())
+        pi_name_email = infodict.pop('principal_investigator')
+        pi_name, pi_email, pi_affiliation = pi_name_email.split(',', 2)
+        infodict['principal_investigator_id'] = StudyPerson.create(
+            pi_name.strip(), pi_email.strip(), pi_affiliation.strip())
 
-    return Study.create(User(owner), title, efo_ids, infodict)
+        return Study.create(User(owner), title, efo_ids, infodict)
```

Comment:
should this return be outside the TRN? I guess it doesn't matter but rather ask ...